### PR TITLE
Tls listener and proxy support

### PR DESCRIPTION
**Pull request by btoonk**

### CODE REVIEW

**Comment by hellt** [comment]
Hi @atoonk @btoonk 
thanks a million for a quick turnaround
I checked branch and it WORKS in my env where external ssh is blocked, but http proxy allows passage of ssh.

File: cmd/client.go [hunk 1]
```diff
@@ -465,7 +466,30 @@ var clientTlsCmd = &cobra.Command{
...
+
+			var wg sync.WaitGroup
+			wg.Add(1)
+
+			go func() {
+				defer wg.Done()
+				log.Printf("Waiting for connection...")
+				lcon, err := l.Accept()
+				if err != nil {
+					log.Printf("Listener: Accept Error: %s\n", err)
```
Comment (lucasallan) [inline cmd/client.go:483]: Should we change it to `Fatalf`? the line below will try to close the connection when the function returns `lcon` might be nil at that point.

Comment (btoonk) [reply cmd/client.go]: You're right, I changed it to fatalf.

**Comment by atoonk** [comment]
Thanks for adding support for:
1) tunnel SSH through a proxy using ```--proxy http://localhost:3128``` 
example: ```mysocketctl tunnel connect -s ca58c819-7ca2-4f81-b6c8-649a2d72c032 -t 9356f1e5-bd89-4476-81de-4c5934760a11 -p 45331 --proxy http://localhost:3128```
2) adding support for a local listener, so we can tunnel traffic from localhost to a TLS socket. example:
```mysocketctl client tls --host polished-wind-2914.edge.mysocket.io -p 31010 -l 3306```
This creates a localhost listener on port 3306. All traffic on that port is forwarded to the TLS socket for polished-wind-2914.edge.mysocket.io.  This is useful for example for database connection through mysocket.

**Review by atoonk — APPROVED** [review]
awesome!

**Pull request merged**